modules/gapi/include/opencv2/gapi/imgproc.hpp

```diff
@@ -110,20 +113,86 @@ namespace imgproc {
         }
     };
 
-    G_TYPED_KERNEL(GCanny, <GMat(GMat,double,double,int,bool)>, "org.opencv.imgproc.canny"){
+    G_TYPED_KERNEL(GCanny, <GMat(GMat,double,double,int,bool)>, "org.opencv.imgproc.feature.canny"){
         static GMatDesc outMeta(GMatDesc in, double, double, int, bool) {
             return in.withType(CV_8U, 1);
         }
     };
 
     G_TYPED_KERNEL(GGoodFeatures,
                    <cv::GArray<cv::Point2f>(GMat,int,double,double,Mat,int,bool,double)>,
-                   "org.opencv.imgproc.goodFeaturesToTrack") {
+                   "org.opencv.imgproc.feature.goodFeaturesToTrack") {
         static GArrayDesc outMeta(GMatDesc, int, double, double, const Mat&, int, bool, double) {
             return empty_array_desc();
         }
     };
 
+    G_TYPED_KERNEL(GFindContours,<GArray<GArray<Point>>(GMat,int,int,Point)>,
+                   "org.opencv.imgproc.shape.findContours")
+    {
+        static GArrayDesc outMeta(GMatDesc in, int mode, int, Point)
+        {
+            GAPI_Assert(in.chan == 1);
+            switch (mode)
+            {
+            case RETR_CCOMP:
+                GAPI_Assert(in.depth == CV_8U || in.depth == CV_32S);
+                break;
+            case RETR_FLOODFILL:
+                GAPI_Assert(in.depth == CV_32S);
+                break;
+            default:
+                GAPI_Assert(in.depth == CV_8U);
+                break;
+            }
+            return empty_array_desc();
+        }
+    };
+
+    G_TYPED_KERNEL(GFindContoursHierarchical,<GFindContoursOutput(GMat,int,int,Point)>,
+                   "org.opencv.imgproc.shape.findContoursHierarchical")
+    {
+        static std::tuple<GArrayDesc,GArrayDesc> outMeta(GMatDesc in, int mode, int, Point)
+        {
+            GAPI_Assert(in.chan == 1);
+            switch (mode)
+            {
+            case RETR_CCOMP:
+                GAPI_Assert(in.depth == CV_8U || in.depth == CV_32S);
+                break;
+            case RETR_FLOODFILL:
+                GAPI_Assert(in.depth == CV_32S);
+                break;
+            default:
+                GAPI_Assert(in.depth == CV_8U);
+                break;
+            }
+            return std::make_tuple(empty_array_desc(), empty_array_desc());
+        }
+    };
+
+    G_TYPED_KERNEL(GBoundingRectMat, <GOpaque<Rect>(GMat)>,
+                   "org.opencv.imgproc.shape.boundingrectMat") {
```

Comment:
Should this and below string identifiers contain Rect with first letter in upper case?
```"org.opencv.imgproc.shape.boundingrectMat" -> "org.opencv.imgproc.shape.boundingRectMat"?```

Reply:
Yes, I think it will be more consistent that way. 
Changed